src/dms/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -207,17 +207,25 @@ TraceId traceId
     /// <summary>
     /// Insert a single Document into the database and return the Id of the new document
     /// </summary>
-    public async Task<long> InsertDocument(
+    public async Task<long> InsertDocumentAndAlias(
         Document document,
+        int referentialPartitionKey,
+        Guid referentialId,
         NpgsqlConnection connection,
         NpgsqlTransaction transaction,
         TraceId traceId
     )
     {
         await using var command = new NpgsqlCommand(
-            @"INSERT INTO dms.Document (DocumentPartitionKey, DocumentUuid, ResourceName, ResourceVersion, IsDescriptor, ProjectName, EdfiDoc, LastModifiedTraceId)
+            @"
+            WITH Documents AS (
+            INSERT INTO dms.Document (DocumentPartitionKey, DocumentUuid, ResourceName, ResourceVersion, IsDescriptor, ProjectName, EdfiDoc, LastModifiedTraceId)
               VALUES ($1, $2, $3, $4, $5, $6, $7, $8)
-              RETURNING Id;",
+              RETURNING Id
```

Comment:
That is a new and creative use for a CTE! I've never seen an insert statement embedded inside a CTE. That's awesome!